### ISSUE TITLE
How can I train YOLO

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/ultralytics-test#14]
I have no GPUs.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

It sounds like you are working without GPUs. Don’t worry — YOLO is designed to work on both GPU and CPU environments. While GPUs provide faster training, you can train models on CPUs as well. For this setup, ensure you have the latest version installed and properly configured. 😊

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics-test/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in environments optimized for both GPU and CPU. Here are some recommendations for training, even without GPU access:
- **Notebooks** with free GPU/CPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Training on local **CPU-only environment**. If you have limited resources, you can reduce batch sizes and training configurations to match your system's capabilities.
- Explore cloud options like **Google Colab** or **Paperspace** for free training GPU runtimes.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will assist you soon if additional help is required. Thanks for your patience! 🚀
